src/core/unit.h

```diff
@@ -464,6 +464,9 @@ typedef struct Unit {
         /* When writing transient unit files, stores which section we stored last. If < 0, we didn't write any yet. If
          * == 0 we are in the [Unit] section, if > 0 we are in the unit type-specific section. */
         signed int last_section_private:2;
+
+        /* Remember this unit was part of some dependency cycle. */
+        bool was_on_dependency_cycle;
```

Comment:
We have lots of units. Bitmaps are used for other bools. Please move this up to the sections with bools and add `:1`.

Reply:
I want to expose this information on dbus as a property (using `SD_BUS_PROPERTY`) and if this becomes a bit in the bitfield then I get compilation error because I need to be able to take address of the variable which is not possible with bitfields.